### PR TITLE
fixed type of `either` in `src/lib/regex.js`

### PR DESCRIPTION
**Pull request by jaacko-torus**
and added types to `types.d.ts`

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -23,6 +23,15 @@ declare module 'highlight.js' {
...
     }
 
+    // perhaps make this an interface?
+    type RegexOptions = {
+        capture?: boolean
+    }
+
+    type RegexEitherArgs =
+        (RegExp | string)[] | [RegexOptions] |
+        [...(RegExp | string)[], RegexOptions]
```
Comment (joshgoebel) [inline types/index.d.ts:33]: You can't have just options... so minimally this should be:

```
  (RegExp | string)[] | 
  [...(RegExp | string)[], RegexEitherOptions]
```

File: types/index.d.ts [hunk 2]
```diff
@@ -23,6 +23,15 @@ declare module 'highlight.js' {
         install: (vue: any) => void
     }
 
+    // perhaps make this an interface?
+    type RegexOptions = {
```
Comment (joshgoebel) [inline types/index.d.ts:27]:
```suggestion
    type RegexEitherOptions = {
```

File: types/index.d.ts (hunk 1 above)
Comment (joshgoebel) [reply types/index.d.ts:33]: And we can probably just inline this type, I don't think we need a custom type for a single function.

**Comment by jaacko-torus** [comment]
That should fix it :D

**Review by joshgoebel — APPROVED** [review]

**Pull request merged**